scripts/fetch-from-github/fetch.js

```diff
@@ -0,0 +1,95 @@
+const fs = require('fs');
+const path = require('path');
+
+// Function to load Octokit once
+let octokitInstance = null;
+
+async function loadOctokit() {
+  if (!octokitInstance) {
+    const { Octokit } = await import('@octokit/rest');
+    octokitInstance = process.env.VBOT_GITHUB_API_TOKEN
+      ? new Octokit({
+          auth: process.env.VBOT_GITHUB_API_TOKEN,
+        })
+      : new Octokit();
+  }
+  return octokitInstance;
+}
+
+const args = process.argv.slice(2);
+
+function printHelp() {
+  console.log(`
+Usage: node script.js <owner> <repo> <path> <save-directory> [custom-filename]
+
+Arguments:
+  <owner>           The GitHub repository owner or organization.
+  <repo>            The GitHub repository name.
+  <path>            The file or directory path within the repository to fetch.
+  <save-directory>  The local directory where files will be saved.
+  [custom-filename] Optional. If provided, the fetched file will be saved with this name.
+
+Options:
+  -h, --help        Show this help message.
+
+Example:
+  node fetch.js redpanda-data cloudv2 proto/gen/openapi/openapi.prod.yaml ./modules/ROOT/attachments cloud-api.yaml
+
+Note: Ensure that you have set the 'VBOT_GITHUB_API_TOKEN' environment variable for authentication.
+  `);
+}
+
+if (args.includes('--help') || args.includes('-h')) {
+  printHelp();
+  process.exit(0);
+}
+
+if (args.length < 4) {
+  console.error('Error: Missing arguments.');
```

Comment:
not sure if mixing two different log levels in the same output is a good idea. Maybe wrap this in another function and call something like 

```
// Log function to print messages at different levels
function logMessage(message, level = 'log') {
  if (level === 'error') {
    console.error(message);
  } else {
    console.log(message);
  }
}

// Print help message function
function printHelp() {
  logMessage(`...`) 
  
  }
...

// Validate the number of arguments provided
if (args.length < 4) {
  logMessage('Error: Missing arguments.', 'error'); // Use 'error' level for error messages
  printHelp();
  process.exit(1);
}
```